Classes/EventStore/EventStore.php

```diff
@@ -52,6 +55,22 @@ public function get(EventStreamFilterInterface $filter): EventStream
         return $eventStream;
     }
 
+    /**
+     * Returns the first EventAndRawEvent instance that matches the given $filter - or throws an EventNotFoundException if no matching events could be found
+     *
+     * @param EventStreamFilterInterface $filter
+     * @return EventAndRawEvent
+     * @throws EventNotFoundException
+     */
+    public function getOne(EventStreamFilterInterface $filter): EventAndRawEvent
```

Comment:
I'd rather prefer not having `EventAndRawEvent` part of the EventStore API. It's okay for the EventStream to return them, but the Store should only be about streams IMO.

Reply:
So returning a stream of `EventAndRawEvent` is OK, but not a single instance?

Reply:
Yeah, because `EventStream` is a much cleaner return type than `EventAndRawEvent`, and we should maybe keep the dirty needs to the lowest levels. Like that we move that dirty type up into the EventStore.

Reply:
Sorry, but I still can't comprehend why `EventStream<EventAndRawEvent>` should be any cleaner than `EventAndRawEvent`. Or what do you mean by "much cleaner"?

Reply:
Sorry for being a PITA. It's really just, that I think the EventStore should only have the API `get(Filter): EventStream` and that EventStream in turn can contain methods to extract the domain events or the raw events.

Reply:
No need for excuses, I value your feedback! But I'm not convinced by the "I think [it] should only have the API [...]" to be honest ;)
Let's see what others think